src/lib/random_oracle_input/random_oracle_input.ml

```diff
@@ -58,20 +58,22 @@ module Chunked = struct
   *)
   let pack_to_fields (type t) (module F : Field_intf with type t = t)
       ~(pow2 : int -> t) { field_elements; packeds } =
-    let shift_left acc n = F.( * ) acc (pow2 n) in
-    let open F in
-    let packed_bits =
-      let xs, acc, acc_n =
-        Array.fold packeds ~init:([], zero, 0)
-          ~f:(fun (xs, acc, acc_n) (x, n) ->
-            let n' = Int.(n + acc_n) in
-            if Int.(n' < size_in_bits) then (xs, shift_left acc n + x, n')
-            else (acc :: xs, zero, 0) )
+    if Array.length packeds = 0 then field_elements
```

Comment:
This feels like unnecessary optimisation. All of the `Array` functions below should already default to a fast path for empty arrays

Reply:
I added this not as optimization but because in the else branch, I always add one field element

Reply:
Let's not do that